schema/encoding/jsonschema/benchmark_complex_test.go

```diff
@@ -0,0 +1,199 @@
+package jsonschema_test
+
+import (
+	"github.com/rs/rest-layer/schema"
+	"math"
+	"time"
+)
+
+var (
+	rfc3339NanoDefault = schema.Field{
+		Description: "UTC start time in RFC3339 format with Nano second support, e.g. 2006-01-02T15:04:05.999999999Z",
+		Validator: &schema.Time{
+			TimeLayouts: []string{time.RFC3339Nano},
+		},
+	}
+)
+
+// Default modifies an existing schema.Field with a default value.
+func Default(f schema.Field, v interface{}) schema.Field {
+	f.Default = v
+	return f
+}
+
+// Required modifies an existing schema.Field to be required.
+func Required(s schema.Field) schema.Field {
+	s.Required = true
+	return s
+}
+
+// RFC3339Nano schema.Field template
+func RFC3339Nano() schema.Field {
+	return rfc3339NanoDefault
+}
+
+// String field
+func String(min, max int, description string) schema.Field {
+	return schema.Field{
+		Description: description,
+		Default:     "",
+		Validator: &schema.String{
+			MinLen: min,
+			MaxLen: max,
+		},
+	}
+}
+
+// Integer returns schema.Field template for int validation.
+func Integer(min, max int, description string) schema.Field {
+	return schema.Field{
+		Default:     min,
+		Description: description,
+		Validator: &schema.Integer{
+			Boundaries: &schema.Boundaries{
+				Min: float64(min),
+				Max: float64(max),
+			},
+		},
+	}
+}
+
+// Bool validator helper
+func Bool(description string) schema.Field {
+	return schema.Field{
+		Description: description,
+		Validator:   &schema.Bool{},
+	}
+}
+
+func getComplexSchema() (schema.Schema, schema.Schema) {
```

Comment:
I where not convened with the so called "requestSchema" and "responseSchema" polluting the namespace but about all the intermediate ones.

The suggested names complexSchema1/2 could be useful for reuse in tests.

> getComplexSchema

Even test code should be clean. Perhaps even cleaner than the main code, since tests also serve as examples/documentation to package maintainers and users.

I believe a good clean function should do one and only one thing. I think this mwthod does two which makes it unclean, and it makes it harder than it has to be to reuse the complex schema for tests